docs/RATE_LIMITING.md

```diff
@@ -0,0 +1,644 @@
+# Hockeypuck Rate Limiting
```

Comment:
Historically we haven't had much in the way of documentation in this repo, it's mainly been in the hockeypuck.io repo. I'm wary of maintaining docs in both places, and have also been bitten a few times by people using HIPs as documentation (because they show up in search results). Need to think more long term about how we maintain docs and whether they should be consolidated. 🤔